src/Core/Addon/Theme/ThemeRepository.php

```diff
@@ -51,22 +51,27 @@ public function __construct(ConfigurationInterface $configuration, Filesystem $f
 
     public function getInstanceByName($name)
     {
-        $dir = $this->appConfiguration->get('_PS_ALL_THEMES_DIR_').$name;
-        $themeCachePath = $this->appConfiguration->get(
-                '_PS_CACHE_DIR_'
-            ) . 'themes/' . $name;
+        $dir = $this->appConfiguration->get('_PS_ALL_THEMES_DIR_') . $name;
 
+        $confDir = $this->appConfiguration->get('_PS_CONFIG_DIR_') . 'themes/' . $name;
+        $jsonConf = '';
         if ($this->shop) {
-            $jsonConf = $themeCachePath . '/shop' . $this->shop->id . '.json';
-        } else {
-            $jsonConf = $themeCachePath . '/theme.json';
```

Comment:
not needed anymore?

Reply:
Replaced behavior. Look at line 71 of the new version.